### PR TITLE
sql: update privilege and user account management info

### PR DESCRIPTION
**Pull request by CaitinChen**
Via
https://github.com/pingcap/docs-cn/pull/1192
https://github.com/pingcap/docs-cn/pull/1257
https://github.com/pingcap/docs-cn/pull/1259
@lilin90 @tiancaiamao PTAL

### CODE REVIEW

File: sql/privilege.md [hunk 1]
```diff
@@ -228,9 +161,124 @@ To be more precise, you can check the privilege information in the `Grant` table
...
+
+### CREATE ROLE/USER
+
+`CREATE ROLE` requires the `CREATE ROLE` privilege.
+
+`CREATE USER` requires the `CREATE USER` privilege.
+
+### DROP ROLE/USER
+
+`DROP ROLE` requires the `DROPROLE` privilege.
```
Comment (tiancaiamao) [inline sql/privilege.md]: s/DROPROLE/DROP ROLE

File: sql/privilege.md [hunk 2]
```diff
@@ -228,9 +161,124 @@ To be more precise, you can check the privilege information in the `Grant` table
...
+
+`DROP USER` requires the `CREATEUSER` privilege.
```
Comment (tiancaiamao) [inline sql/privilege.md]: s/CREATEUSER/CREATE USER

File: sql/privilege.md [hunk 3]
```diff
@@ -228,9 +161,124 @@ To be more precise, you can check the privilege information in the `Grant` table
...
+
+### ALTER USER
+
+Requires the `CREATEUSER` privilege.
```
Comment (tiancaiamao) [inline sql/privilege.md]: s/CREATEUSER/CREATE USER

File: sql/privilege.md [hunk 4]
```diff
@@ -228,9 +161,124 @@ To be more precise, you can check the privilege information in the `Grant` table
...
+| Show View      | `ShowViewPriv`   | Executes `SHOW CREATE VIEW`    |
+| References     | `ReferencesPriv` | /                |
+| Create View    | `CreateViewPriv` | Creates a View                 |
+| Create User    | `CreateUserPriv` | Creates a user                |
+| Create Role    | `CreateRolePriv` | Executes `CREATE ROLE`         |
+| Show Databases | `ShowDBPriv`     | Shows the table status in the database |
+
+### ALTER
+
+- For all the `ALTER` statement, users must have the `ALTER` privilege for the corresponding table.
```
Comment (lilin90) [inline sql/privilege.md]:
```suggestion
- For all `ALTER` statements, users must have the `ALTER` privilege for the corresponding table.
```

File: sql/privilege.md [hunk 5]
```diff
@@ -228,9 +161,124 @@ To be more precise, you can check the privilege information in the `Grant` table
...
+
+To execute the `CREATE TABLE...LIKE...` statement, the `SELECT` privilege for the table is required.
+
+### CREATE VIEW
+
+Requires the `CREATE VIEW` privilege.
+
+> **Note:**
+>
+> If the current user is not the user that creates the View, he needs both the `CREATE VIEW` and `SUPER` privileges.
```
Comment (lilin90) [inline sql/privilege.md]:
```suggestion
> If the current user is not the user that creates the View, both the `CREATE VIEW` and `SUPER` privileges are required.
```

File: sql/privilege.md [hunk 6]
```diff
@@ -228,9 +161,124 @@ To be more precise, you can check the privilege information in the `Grant` table
...
+
+Requires the `ALTER` and `DROP` privileges for the table before renaming and the `CREATE` and `INSERT` privileges for the table after renaming.
+
+### ANALYZE TABLE
+
+Requires the `INSERT` and `SELECT` privileges for the table.
+
+### SHOW
+
+`SHOW CREATE TABLE` requires any of the privileges.
```
Comment (lilin90) [inline sql/privilege.md]: Does this mean all privileges or any one of the privileges?

File: sql/user-account-management.md [hunk 7]
```diff
@@ -76,23 +76,44 @@ Currently, TiDB does not support setting account resource limits.
...
-```
+    ```sql
+    ALTER USER 'jeffrey'@'localhost' IDENTIFIED BY 'mypass';
+    ```
+
+## Forget the `root` password
+
+1. Modify the configuration file by adding `skip-grant-table` in the `security` part:
+
+    > [security]
```
Comment (lilin90) [inline sql/user-account-management.md]:
```suggestion
    ```
    [security]
```

File: sql/user-account-management.md [hunk 8]
```diff
@@ -76,23 +76,44 @@ Currently, TiDB does not support setting account resource limits.
...
+    > skip-grant-table = true
```
Comment (lilin90) [inline sql/user-account-management.md]:
```suggestion
    skip-grant-table = true
    ```
```

File: sql/privilege.md (hunk 6 above)
Comment (morgo) [inline sql/privilege.md]:
```suggestion
`SHOW CREATE TABLE` requires any single privilege to the table.
```

File: sql/privilege.md [hunk 9]
```diff
@@ -6,80 +6,13 @@ category: user guide
 
 # Privilege Management
 
-TiDB's privilege management system is implemented according to the privilege management system in MySQL. It supports most of the syntaxes and privilege types in MySQL. If you find any inconsistency with MySQL, feel free to [open an issue](https://github.com/pingcap/tidb/issues/new/choose).
+TiDB's privilege management system is implemented according to the privilege management system in MySQL. It supports most of the syntaxes and privilege types in MySQL.
```
Comment (morgo) [inline sql/privilege.md]: I think it's important to nuance what the support looks like. TiDB is mostly compatible with MySQL 5.7, but supports roles as well (MySQL 8.0).  It does not support [dynamic privileges](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#privileges-provided-dynamic) (8.0) to my knowledge.

```suggestion
TiDB supports MySQL 5.7's privilege management system, including the syntax and privilege types. Starting with TiDB 3.0, support for SQL Roles is also available.
```

PTAL @tiancaiamao

Comment (tiancaiamao) [reply sql/privilege.md]: Yes, you're right

File: sql/privilege.md (hunk 6 above)
Comment (CaitinChen) [reply sql/privilege.md]: @lilin90 Any one of the privileges.

**Comment by tiancaiamao** [comment]
LGTM

File: sql/privilege.md [hunk 10]
```diff
@@ -291,7 +339,7 @@ Data in the `user` and `db` tables is also sorted when loaded into memory.
 
 The use of `%` in `tables_priv` and `columns_priv` is similar, but column value in `Db`, `Table_name` and `Column_name` cannot contain `%`. The sorting is also similar when loaded.
 
-#### Time of effect
+### Time of effect
 
 When TiDB starts, some privilege-check tables are loaded into memory, and then the cached data is used to verify the privileges. The system will periodically synchronize the `grant` table from database to cache. Time of effect is determined by the synchronization cycle. Currently, the value is 5 minutes.
```
Comment (morgo) [inline sql/privilege.md]: This statement is now out of date. Using the privilege management statement interface will [synchronously run flush privileges](https://github.com/pingcap/tidb/pull/8886) (for mysql compatibility).

```suggestion
When TiDB starts, some privilege-check tables are loaded into memory, and then the cached data is used to verify the privileges. Executing privilege management statements such as `GRANT`, `REVOKE`, `CREATE USER`, `DROP USER` will take effect immediately.

Manually editing tables such as `mysql.user` with statements such as `INSERT`, `DELETE`, `UPDATE` will not take effect immediately. This behavior is compatible with MySQL, and privilege cache can be updated with the following statement:
```

PTAL @tiancaiamao

**Comment by CaitinChen** [comment]
@tiancaiamao PTAL

**Comment by CaitinChen** [comment]
@lilin90 PTAL again.

**Comment by tiancaiamao** [comment]
LGTM

File: sql/user-account-management.md [hunk 11]
```diff
@@ -76,23 +76,46 @@ Currently, TiDB does not support setting account resource limits.
...
+    skip-grant-table = true
+    ```
+
+2. Restart TiDB using the modified configuration (the `root` privilege is required):
+
+    ```bash
+    sudo ./tidb-server -skip-grant-table=true -store=tikv -path=...
+    ```
+
+    By this configuration parameter, TiDB skips the privilege system.
```
Comment (lilin90) [inline sql/user-account-management.md]: This sentence is not clear enough. Which parameter? Could you make it clearer?

**Comment by CaitinChen** [comment]
@lilin90 I have confirmed with @tiancaiamao to clarify the "configuration parameter". PTAL again. Thanks.

**Pull request merged**